src/env/mod.rs

```diff
@@ -22,6 +22,8 @@ pub trait FileSystem: Send + Sync {
 
     fn delete<P: AsRef<Path>>(&self, path: P) -> Result<()>;
 
+    fn rename<P: AsRef<Path>>(&self, src_path: P, dst_path: P) -> Result<()>;
```

Comment:
Add a flag `need_data`, when need_data is false, tikv can use a new encryption key.